### PR TITLE
Dependent lookup plugin

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
This adds a new lookup plugin, similar to the nested lookup plugin, which allows to compute the lists/dicts in the argument list depending on earlier indices.

##### ISSUE TYPE
New Module Pull Request

##### COMPONENT NAME
dependent

##### ANSIBLE VERSION
```
2.7.0
```

### CODE REVIEW

File: test/integration/targets/lookups/tasks/main.yml [hunk 1]
```diff
@@ -305,3 +305,45 @@
...
+
+# Dependent lookups
+
+- name: Test dependent lookup
+  debug: var=item
+  with_dependent:
+    - a:
+      - 1
+      - 2
+      b:
```
Comment (gundalow) [inline test/integration/targets/lookups/tasks/main.yml:324]: Missing `-`?

Comment (felixfontein) [reply test/integration/targets/lookups/tasks/main.yml:324]: No, that's intended. The first (outer) list item contains a dict, with two keys `a` and `b`.

**Comment by felixfontein** [comment]
Retriggering tests

**Comment by sivel** [comment]
I'd lean more towards having this as a filter.  We haven't firmly decided on the change, but we have been leaning towards moving away from lookups to provide looping functionality.

**Pull request closed without merging**